src/main/java/batterystreamreceiver/BatteryProperties.java

```diff
@@ -0,0 +1,59 @@
+package batterystreamreceiver;
+
+public class BatteryProperties {
```

Comment:
`BatteryProperties` has only getters and setters. This usually indicates an opportunity for better encapsulation. Why not encapsulate the data together with the computation functionality in this class?